Content.Server/Botany/Systems/SeedExtractorSystem.cs

```diff
@@ -32,8 +32,14 @@ private void OnInteractUsing(EntityUid uid, SeedExtractorComponent component, In
 
         if (TryComp(args.Used, out ProduceComponent? produce))
         {
-            if (!_prototypeManager.TryIndex<SeedPrototype>(produce.SeedName, out var seed))
-                return;
+            SeedPrototype? seed;
+            // try get seed from seed database
+            if (produce.SeedUid == null || !_botanySystem.Seeds.TryGetValue(produce.SeedUid.Value, out seed))
+            {
+                // try get seed from base prototype
+                if (produce.SeedName == null || !_prototypeManager.TryIndex(produce.SeedName, out seed))
+                    return;
+            }
```

Comment:
can this get pulled out into a method somewhere, its used in 3 diff places

Reply:
Moved into a function